scripts/install-build-tools.sh

```diff
@@ -6,19 +6,61 @@ green="\033[0;32m"
 cyan="\033[0;36m"
 end="\033[0m"
 
-set -e
+# for debugging, 'set -x' can be added to trace the commands
+set -ex
 
 SUDO=''
 if (( $EUID != 0 )); then
     echo -e "non-root user, sudo required"
     SUDO='sudo'
 fi
 
+# install in a custom prefix rather than /usr/local. by default, this
+# chooses "prefix" directory alongside "scripts" directory.
+PREFIX="$(cd "$(dirname "$0")"/.. && pwd)/prefix"
+echo "Will install local dependencies in the following prefix: $PREFIX"
+mkdir -p $PREFIX $PREFIX/bin $PREFIX/lib $PREFIX/include
+
+# Supporting these versions for buildflow
+PYTHON_VERSIONS=("3.12" "3.11" "3.10")
+PYTHON_PACKAGES=("setuptools" "matplotlib" "numpy" "pylint")
+PY_INSTALLED=''
+echo "Python3 versions supported: ${PYTHON_VERSIONS[@]}"
+
+# let user call python scripts as ./usr/bin/python scripts/script.py
+make_symlinks() {
+  PY_LOCATION=$1
+  PREFIX_BIN=${PREFIX}/bin
```

Comment:
Please quote all variable expansions. Someone will inevitably use a path with a space in it, and that will likely explode fantastically.

Reply:
ACK - quoted all paths, would be helpful to see your review if they are valid now

Reply:
It would be far better to quote all variable expansions rather than just all paths. This is a good shell-scripting habit to get into early.